Acknowledge terminal width for output on Unix-systems

Veykril opened the pull request:
This PR implements #160 for the unix family, changing the output width from 79 to the terminal width if it is wider than 79. I pulled in libc as a dependency for the syscall to make. I will implement the windows variant as well, but I wanted to know if it is okay to use the winapi crate for this or if we should define the one or two functions that are required in the `main.rs` ourself.

Also this PR changes the default behaviour. I wasnt sure if this should be an argument or not. If it should be I'd gladly add that as well if you propose a definition for it.

Aaronepower commented on src/main.rs:
```diff
@@ -201,35 +203,37 @@ fn main() {
 
-    print_language(&total, "Total");
-    println!("{}", ROW);
+    print_language(columns - 61, &total, "Total");
```
What is 61? This should probably be a `const` instead of a magic number.

Aaronepower commented on src/main.rs:
```diff
@@ -201,35 +203,37 @@ fn main() {
-fn print_results<'a, I>(languages: I, list_files: bool)
+fn print_results<'a, I>(row: &str, languages: I, list_files: bool)
 where I: std::iter::Iterator<Item = (&'a LanguageType, &'a Language)> {
+    let path_len = row.len() - 54;
```
Same as above about magic numbers.

Veykril commented:
Added a winapi version and changed the magic numbers to constants. I hope the names for them are okay cause I wasnt really sure what to call them. They both are the number of characters in a row that exclude the language/path section with and without the spaces between the sections counted. Maybe you can come up with a better name for them, and on that note they probably shouldnt contain lang but path in their names instead.

Aaronepower commented on src/main.rs:
```diff
@@ -176,16 +179,19 @@ fn main() {
-    println!("{}", ROW);
-
+                "Blanks",
+                columns - 67);
```
There is another magic number here.

Aaronepower commented:
Hmm, looking at this code with the `winapi` version makes me wonder if `terminal_columns` should actually be a seperate crate, as I think this would help a lot of people writing CLI apps. How would you feel about pulling out that code and making it a crate?

Veykril commented:
Sounds like a good idea to explore, a small crate with helper functions like that. I'll experiment with that a bit and come back to this later.

Veykril commented:
After searching a bit more it turns out there is already such a [crate](https://github.com/clap-rs/term_size-rs), I changed the implementation to rely on this instead.

Aaronepower commented on src/stats.rs:
```diff
@@ -72,13 +73,15 @@ impl fmt::Display for Stats {
         } else {
             let mut formatted = String::from("|");
-            let from = find_char_boundary(&name, name_length - 24);
+            let from = find_char_boundary(&name, name_length + 1 - max_len);
```
Can you put comment here explaining the `+ 1`?

Veykril commented:
Does this explain it properly?

Aaronepower commented:
@Veykril Yeah looks good to me, once the CI has completed I'll merge it in.
